### ISSUE TITLE
SQLite trigger with RAISE(ROLLBACK, ...) does not parse

### ISSUE THREAD

**Issue by kgutwin** [sqlfluff/sqlfluff#6101]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The SQLite dialect cannot parse the [raise function syntax](https://www.sqlite.org/syntax/raise-function.html), valid [within triggers](https://www.sqlite.org/lang_createtrigger.html#the_raise_function):

```sql
CREATE TRIGGER x
BEFORE UPDATE OF z ON y
BEGIN
SELECT RAISE(ROLLBACK, 'updating z not allowed');
END;
```

Quoting the bare word `ROLLBACK` allows parsing to proceed, but that is a syntax error for SQLite.

### Expected Behaviour

```
All Finished 📜 🎉!
```

### Observed Behaviour

```
== [stdin] FAIL                                                                                                                                                                    
L:   4 | P:  14 |  PRS | Line 4, Position 14: Found unparsable section:
                       | "ROLLBACK, 'updating z not allowed'"
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```
echo "CREATE TRIGGER x
BEFORE UPDATE OF z ON y
BEGIN
SELECT RAISE(ROLLBACK, 'updating z not allowed'); 
END;" | sqlfluff lint --dialect sqlite -
```

### Dialect

sqlite

### Version

sqlfluff, version 3.1.0
Python 3.11.9

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
